### ISSUE TITLE
Why logging with scaled loss: loss_sum / sample_size / math.log(2)

### ISSUE THREAD

**Issue by Epsilon-Lee** [pytorch/fairseq#466]
In cross_entropy.py file, you define an aggregate method to prepare training statistics for logging as follows:
```python
@staticmethod
    def aggregate_logging_outputs(logging_outputs):
        """Aggregate logging outputs from data parallel training."""
        loss_sum = sum(log.get('loss', 0) for log in logging_outputs)
        ntokens = sum(log.get('ntokens', 0) for log in logging_outputs)
        nsentences = sum(log.get('nsentences', 0) for log in logging_outputs)
        sample_size = sum(log.get('sample_size', 0) for log in logging_outputs)
        agg_output = {
            'loss': loss_sum / sample_size / math.log(2),
            'ntokens': ntokens,
            'nsentences': nsentences,
            'sample_size': sample_size,
        }
        if sample_size != ntokens:
            agg_output['nll_loss'] = loss_sum / ntokens / math.log(2)
        return agg_output
```
I wonder why you scale the `loss`(`nll_loss`) by a factor of `1/math.log(2)`, is that for the reason when using fp16 training, the loss will be too small? Many thanks!

**Comment by myleott** [comment]
It's just to convert from base e to base 2, since perplexity is typically reported in terms of bits.
